### ISSUE TITLE
Some Linux flavors fail to dlopen

### ISSUE THREAD

**Issue by maluoi** [StereoKit/StereoKit#485]
## Description
From gamethecupdog and ArgentumIpsum on Discord

StereoKit is failing to load libStereoKitC.so during initialization, and then later crashing when getting to C#'s own PInvoke code. (StereoKit needs to manually load the library before C# tries to PInvoke, as .net standard 2.0 doesn't have good enough tools to discover the library location on its own.)

## Platform / Environment

C#
v0.3.6
Linux: Fedora, Mint, and an unspecified 'Arch based'

## Logs or exception details

```
[SK error] Failed to load StereoKitC!
Unhandled exception. System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.DllNotFoundException: Unable to load shared library 'StereoKitC' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libStereoKitC: cannot open shared object file: No such file or directory
   at StereoKit.NativeAPI.time_scale(Double scale)
   at StereoKit.Time.set_Scale(Double value) in C:\Data\Repositories\Stereo Kit\StereoKit\Util\Time.cs:line 27
   at App..ctor(String[] args) in /home/[name]/Documents/VR/stereokit/StereoKit/Examples/StereoKitTest/App.cs:line 46
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Span`1& arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.RuntimeType.CreateInstanceImpl(BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture)
   at System.Activator.CreateInstance(Type type, BindingFlags bindingAttr, Binder binder, Object[] args, CultureInfo culture, Object[] activationAttributes)
   at System.Activator.CreateInstance(Type type, Object[] args)
   at Program.Main(String[] args) in /home/[name]/Documents/VR/stereokit/StereoKit/Examples/StereoKitTest/Program.cs:line 15
```
with LD_DEBUG
```
     11097:     find library=libStereoKitC.so [0]; searching
     11097:      search cache=/etc/ld.so.cache
     11097:      search path=/usr/lib           (system search path)
     11097:       trying file=/usr/lib/libStereoKitC.so
     11097:
     11097:     find library=StereoKitC [0]; searching
     11097:      search cache=/etc/ld.so.cache
     11097:      search path=/usr/lib           (system search path)
     11097:       trying file=/usr/lib/StereoKitC
     11097:
     11097:     find library=libStereoKitC [0]; searching
     11097:      search cache=/etc/ld.so.cache
     11097:      search path=/usr/lib           (system search path)
     11097:       trying file=/usr/lib/libStereoKitC
```

```
     13614:     find library=libX11.so.6 [0]; searching
     13614:      search path=/home/[name]/Documents/VR/stereokit/StereoKit/Examples/StereoKitTest/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     13614:       trying file=/home/[name]/Documents/VR/stereokit/StereoKit/Examples/StereoKitTest/bin/Debug/net6.0/runtimes/linux-x64/native/libX11.so.6
     13614:      search cache=/etc/ld.so.cache
     13614:       trying file=/usr/lib/libX11.so.6
     13614:
     13614:     find library=libjsoncpp.so.1 [0]; searching
     13614:      search path=/home/[name]/Documents/VR/stereokit/StereoKit/Examples/StereoKitTest/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     13614:       trying file=/home/[name]/Documents/VR/stereokit/StereoKit/Examples/StereoKitTest/bin/Debug/net6.0/runtimes/linux-x64/native/libjsoncpp.so.1
     13614:      search cache=/etc/ld.so.cache
     13614:      search path=/usr/lib           (system search path)
     13614:       trying file=/usr/lib/libjsoncpp.so.1
     13614:
[SK error] Failed to load StereoKitC!
     13614:     find library=StereoKitC.so [0]; searching
     13614:      search path=/home/[name]/Documents/VR/stereokit/StereoKit/Examples/StereoKitTest/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     13614:       trying file=/home/[name]/Documents/VR/stereokit/StereoKit/Examples/StereoKitTest/bin/Debug/net6.0/runtimes/linux-x64/native/StereoKitC.so
     13614:      search cache=/etc/ld.so.cache
     13614:      search path=/usr/lib           (system search path)
     13614:       trying file=/usr/lib/StereoKitC.so
```

**Comment by ArgentumIpsum** [comment]
Here are my logs of this issue on fedora:

```[SK error] Failed to load StereoKitC!
Unhandled exception. System.DllNotFoundException: Unable to load shared library 'StereoKitC' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libStereoKitC: cannot open shared object file: No such file or directory
   at StereoKit.NativeAPI.sk_run(Action app_update, Action app_shutdown)
   at StereoKit.SK.Run(Action onStep, Action onShutdown) in C:\Data\Repositories\Stereo Kit\StereoKit\SK.cs:line 199
   at Program.<Main>$(String[] args) in /var/home/[name]/Documents/VSCode/HelloSK/Program.cs:line 4
```

```     10252:     find library=libGL.so.1 [0]; searching
     10252:      search cache=/etc/ld.so.cache
     10252:       trying file=/lib64/libGL.so.1
     10252:
     10252:     find library=libGLEW.so.2.1 [0]; searching
     10252:      search cache=/etc/ld.so.cache
     10252:      search path=/lib64:/usr/lib64          (system search path)
     10252:       trying file=/lib64/libGLEW.so.2.1
     10252:       trying file=/usr/lib64/libGLEW.so.2.1
     10252:
     10252:     find library=libStereoKitC.so [0]; searching
     10252:      search cache=/etc/ld.so.cache
     10252:      search path=/lib64:/usr/lib64          (system search path)
     10252:       trying file=/lib64/libStereoKitC.so
     10252:       trying file=/usr/lib64/libStereoKitC.so
     10252:
[SK error] Failed to load StereoKitC!
     10252:     find library=StereoKitC.so [0]; searching
     10252:      search cache=/etc/ld.so.cache
     10252:      search path=/lib64:/usr/lib64          (system search path)
     10252:       trying file=/lib64/StereoKitC.so
     10252:       trying file=/usr/lib64/StereoKitC.so
     10252:
     10252:     find library=libGL.so.1 [0]; searching
     10252:      search cache=/etc/ld.so.cache
     10252:       trying file=/lib64/libGL.so.1
     10252:
     10252:     find library=libGLEW.so.2.1 [0]; searching
     10252:      search cache=/etc/ld.so.cache
     10252:      search path=/lib64:/usr/lib64          (system search path)
     10252:       trying file=/lib64/libGLEW.so.2.1
     10252:       trying file=/usr/lib64/libGLEW.so.2.1
     10252:
     10252:     find library=libStereoKitC.so [0]; searching
     10252:      search cache=/etc/ld.so.cache
     10252:      search path=/lib64:/usr/lib64          (system search path)
     10252:       trying file=/lib64/libStereoKitC.so
     10252:       trying file=/usr/lib64/libStereoKitC.so
     10252:
     10252:     find library=StereoKitC [0]; searching
     10252:      search cache=/etc/ld.so.cache
     10252:      search path=/lib64:/usr/lib64          (system search path)
     10252:       trying file=/lib64/StereoKitC
     10252:       trying file=/usr/lib64/StereoKitC
     10252:
     10252:     find library=libStereoKitC [0]; searching
     10252:      search cache=/etc/ld.so.cache
     10252:      search path=/lib64:/usr/lib64          (system search path)
     10252:       trying file=/lib64/libStereoKitC
     10252:       trying file=/usr/lib64/libStereoKitC
     10252:
Unhandled exception.
```
Here is how it looks when I set LD_LIBRARY_PATH to the directory the libraries are in:
```     10325:     find library=libGL.so.1 [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/libGL.so.1
     10325:      search cache=/etc/ld.so.cache
     10325:       trying file=/lib64/libGL.so.1
     10325:
     10325:     find library=libGLEW.so.2.1 [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/libGLEW.so.2.1
     10325:      search cache=/etc/ld.so.cache
     10325:      search path=/lib64:/usr/lib64          (system search path)
     10325:       trying file=/lib64/libGLEW.so.2.1
     10325:       trying file=/usr/lib64/libGLEW.so.2.1
     10325:
     10325:     find library=libStereoKitC.so [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/libStereoKitC.so
     10325:
     10325:     find library=libGL.so.1 [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/libGL.so.1
     10325:      search cache=/etc/ld.so.cache
     10325:       trying file=/lib64/libGL.so.1
     10325:
     10325:     find library=libGLEW.so.2.1 [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/libGLEW.so.2.1
     10325:      search cache=/etc/ld.so.cache
     10325:      search path=/lib64:/usr/lib64          (system search path)
     10325:       trying file=/lib64/libGLEW.so.2.1
     10325:       trying file=/usr/lib64/libGLEW.so.2.1
     10325:
[SK error] Failed to load StereoKitC!
     10325:     find library=StereoKitC.so [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/StereoKitC.so
     10325:      search cache=/etc/ld.so.cache
     10325:      search path=/lib64:/usr/lib64          (system search path)
     10325:       trying file=/lib64/StereoKitC.so
     10325:       trying file=/usr/lib64/StereoKitC.so
     10325:
     10325:     find library=libGL.so.1 [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/libGL.so.1
     10325:      search cache=/etc/ld.so.cache
     10325:       trying file=/lib64/libGL.so.1
     10325:
     10325:     find library=libGLEW.so.2.1 [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/libGLEW.so.2.1
     10325:      search cache=/etc/ld.so.cache
     10325:      search path=/lib64:/usr/lib64          (system search path)
     10325:       trying file=/lib64/libGLEW.so.2.1
     10325:       trying file=/usr/lib64/libGLEW.so.2.1
     10325:
     10325:     find library=libStereoKitC.so [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/libStereoKitC.so
     10325:
     10325:     find library=libGL.so.1 [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/libGL.so.1
     10325:      search cache=/etc/ld.so.cache
     10325:       trying file=/lib64/libGL.so.1
     10325:
     10325:     find library=libGLEW.so.2.1 [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/libGLEW.so.2.1
     10325:      search cache=/etc/ld.so.cache
     10325:      search path=/lib64:/usr/lib64          (system search path)
     10325:       trying file=/lib64/libGLEW.so.2.1
     10325:       trying file=/usr/lib64/libGLEW.so.2.1
     10325:
     10325:     find library=StereoKitC [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/StereoKitC
     10325:      search cache=/etc/ld.so.cache
     10325:      search path=/lib64:/usr/lib64          (system search path)
     10325:       trying file=/lib64/StereoKitC
     10325:       trying file=/usr/lib64/StereoKitC
     10325:
     10325:     find library=libStereoKitC [0]; searching
     10325:      search path=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native              (LD_LIBRARY_PATH)
     10325:       trying file=/home/[name]/Documents/VSCode/HelloSK/bin/Debug/net6.0/runtimes/linux-x64/native/libStereoKitC
     10325:      search cache=/etc/ld.so.cache
     10325:      search path=/lib64:/usr/lib64          (system search path)
     10325:       trying file=/lib64/libStereoKitC
     10325:       trying file=/usr/lib64/libStereoKitC
     10325:
Unhandled exception.
```

I'm not sure why it can't find libStereokitC.so, but the StereoKitC shared object seems to be stored as StereoKitC.sym. Renaming it does not fix the issue, so I'm not sure if it's relevant.

**Comment by maluoi** [comment]
A fix for this is now up in v0.3.7-preview.6! If this continues, please re-open!
